src/Tests/dotnet-MsiInstallation.Tests/Framework/VMTestBase.cs

```diff
@@ -94,14 +94,22 @@ protected void DeployStage2Sdk()
                 return;
             }
 
-            var installedSdkFolder = $@"c:\Program Files\dotnet\sdk\{SdkInstallerVersion}";
+            var result = VM.CreateRunCommand("dotnet", "--version")
+                .WithIsReadOnly(true)
+                .Execute();
+
+            result.Should().Pass();
+
+            string existingVersionToOverwrite = result.StdOut;
+
+            var installedSdkFolder = $@"c:\Program Files\dotnet\sdk\{existingVersionToOverwrite}";
```

Comment:
This path is only safe if someone is using an x64 SDK, so this won't work if you ever ran this under x64 emulation on arm64.